### DISCUSSION TITLE
Google/Android Pass Generation ?

### DISCUSSION THREAD

**Discussion by MentalGear** [alexandercerutti/passkit-generator#121] [Q&A] [answered]
Thanks for this highly useful library. Is it also possible to generate Google Pay Passes (Android) ?
If not, what is the best approach to generate cross OS/wallet passes?

**Comment by alexandercerutti** [comment, marked as answer]
Hey @MentalGear, thanks for using passkit-generator.

Sadly not, this is only for Apple Wallet. I tried to dive into Google Wallet passes generation but didn't got all the details and an idea on how they are generated (what I know is that they are sent from Google Servers).

Sorry, but I don't know which are the alternatives for cross-generation.

**Reply by MentalGear** [reply]
Thanks for the quick reply. I think for cross OS support, the easiest thing to do is write a google wallet pass and use https://github.com/google-pay/pass-converter to convert it to an apple pass.

**Reply by spencercap** [reply]
@MentalGear did you end up using pass-converter? wondering how that flow feels

**Comment by MentalGear** [comment]
I didn't continue with the passkit generator, as Apple requires a paid dev account, which is something I won't pay for for a small generic membership card ( usecase: local club). Thx for looking into it though, Alexander!

**Comment by adhyan1991-hue** [comment]
Hi @MentalGear Thanks for the discussion! I just wanted to add some clarity for anyone looking into cross-platform pass generation. As mentioned, passkit-generator works great for Apple Wallet but unfortunately doesn’t support Google Wallet (Android). Generating Google Wallet passes is indeed a bit different because the passes are issued via Google’s servers using their official API.

For cross-platform support, one approach is exactly what @MentalGear suggested: creating a Google Wallet pass first and then using tools like [google-pay/pass-converter](https://github.com/google-pay/pass-converter)
 to convert it into an Apple Wallet pass. This works for simple cases like membership cards, tickets, or coupons. If your use case involves generic membership cards, loyalty programs, or local club passes, there are also third-party platforms like [Membership Anywhere](https://membershipanywhere.com/) that handle digital membership cards, Apple Wallet & Google Wallet tickets, and even loyalty reward programs. They abstract the complexities of APIs, certificates, and server-side requirements, making it easier for small businesses or clubs to issue digital cards without investing in developer accounts or custom integration.

if you’re testing for personal or small-scale projects, remember Apple Wallet does require a paid developer account for distribution. Google Wallet, on the other hand, is free for developers but still needs API integration. Using third-party solutions can save a lot of time and headaches if you want a ready-to-go cross-platform solution.
